portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,13 +37,21 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Decide the max number of comments to display
+    final int maxComments = Integer.parseInt(
+                getParameterWithDefault(request, "comments-to-show", DEFALUT_COMMENTS_TO_SHOW));
+
     // Fetch the comment from the datastore
     PreparedQuery comments = DEFAULT_DATASTORE_SERVICE.prepare(new Query("Comment"));
```

Comment:
consider to use https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.Builder.html#withlimit